tactics/inv.ml

```diff
@@ -118,9 +118,9 @@ let make_inv_predicate env evd indf realargs id status concl =
    In any case, we carry along the rest of pairs *)
   let eqdata =
     try Coqlib.build_coq_eq_data ()
-    with UserError _ ->
+    with Coqlib.NameNotRegistered _ ->
     try Coqlib.build_coq_identity_data ()
-    with UserError _ -> user_err (str "No registered equality.") in
+    with Coqlib.NameNotRegistered _ -> user_err (str "No registered equality.") in
```

Comment:
The message could be more informative here:
```suggestion
    with Coqlib.NameNotRegistered _ -> user_err (str "Missing registration of " ++ str s ++ str ".") in
```
(with the `s` coming from the previous failure)